config/initializers/datadog.rb

```diff
@@ -18,16 +18,8 @@
   c.remote.enabled = enabled
 
   unless enabled
-    # TODO: https://github.com/DataDog/dd-trace-rb/issues/2542
-    # disable log tags loaded super early by ddtrace/auto_instrument
-    # required in Gemfile, since they are polluting development log
-    original_tags = Array.wrap(Rails.application.config.log_tags).reject { |tag| tag&.source_location&.first&.include?('datadog') }
-    Rails.application.config.log_tags = original_tags
-
-    c.tracing.transport_options = proc { |t|
-      # Set transport to no-op mode. Does not retain traces.
-      t.adapter :test
-    }
+    c.tracing.log_injection = false
```

Comment:
This is the fix for the linked issue \#2542.

Reply:
Nice!